File: 05_vetores/ex03_busca_linear.c
/*
Exercicio 03
Tema: 05_vetores
*/

#include <stdio.h>

int busca_linear(int * arr, int tamanho, int target){
    for(int i = 0; i < tamanho; i++){
        if(arr[i] == target)
            return i;
    }
    return -1;
}

int main() {

    int arr[] = {5,3,8,1,7};
    int target = 8;
    int tamanho = sizeof(arr) / sizeof(arr[0]);

    int resultado = busca_linear(arr, tamanho,target);

    if(resultado == -1){
        printf("Elemento nao encontrado\n");
    } else {
        printf("O numero '%d' se encontra na posicao: %d\n", target, resultado);
    }

    return 0;
}

File: 05_vetores/ex03_busca_linear.py
"""
Exercicio 03
Tema: 05_vetores
"""

def busca_linear(arr, tamanho, target):
    for i in range(0, tamanho):
        if arr[i] == target:
            return i
    return -1

arr = [5,3,8,1,7]
target = 8

resultado = busca_linear(arr, len(arr), target)

if resultado == -1:
    print("Elemento nao encontrado")
else:
    print(f"Elemento '{target}' encontrado na posicao {resultado}")
